### ISSUE TITLE
Bearer token has expired

### ISSUE THREAD

**Issue by j-oflaherty** [singer-io/tap-asana#35]
The tasks stream takes over an hour to replicate so the token expires. Is there any way to fix this. Here is the error:
```
CRITICAL No Authorization: The bearer token has expired. If you have a refresh token, please use it to request a new bearer token, otherwise allow the user to re-authenticate.
Traceback (most recent call last):
  File ".virtualenvs/tap-asana/bin/tap-asana", line 8, in <module>
    sys.exit(main())
  File "[working_directory]/.virtualenvs/tap-asana/lib/python3.8/site-packages/singer/utils.py", line 229, in wrapped
    return fnc(*args, **kwargs)
  File "[working_directory]/.virtualenvs/tap-asana/lib/python3.8/site-packages/tap_asana/__init__.py", line 185, in main
    sync()
  File "[working_directory]/.virtualenvs/tap-asana/lib/python3.8/site-packages/tap_asana/__init__.py", line 137, in sync
    for rec in stream.sync():
  File "[working_directory]/.virtualenvs/tap-asana/lib/python3.8/site-packages/tap_asana/streams/base.py", line 157, in sync
    for obj in self.get_objects():
  File "[working_directory]/.virtualenvs/tap-asana/lib/python3.8/site-packages/tap_asana/streams/tasks.py", line 58, in get_objects
    for task in self.call_api("tasks", project=project["gid"], opt_fields=opt_fields, modified_since=modified_since):
  File "[working_directory]/.virtualenvs/tap-asana/lib/python3.8/site-packages/asana/page_iterator.py", line 58, in items
    for page in self:
  File "[working_directory]/.virtualenvs/tap-asana/lib/python3.8/site-packages/asana/page_iterator.py", line 40, in __next__
    result = self.get_initial()
  File "[working_directory]/.virtualenvs/tap-asana/lib/python3.8/site-packages/asana/page_iterator.py", line 69, in get_initial
    return self.client.get(self.path, self.query, **self.options)
  File "[working_directory]/.virtualenvs/tap-asana/lib/python3.8/site-packages/asana/client.py", line 173, in get
    return self.request('get', path, params=query, **options)
  File "[working_directory]/.virtualenvs/tap-asana/lib/python3.8/site-packages/asana/client.py", line 91, in request
    raise STATUS_MAP[response.status_code](response)
asana.error.NoAuthorizationError: No Authorization: The bearer token has expired. If you have a refresh token, please use it to request a new bearer token, otherwise allow the user to re-authenticate.
```

**Comment by ghost** [comment]
Hi Julian! Im expriencing the same error here. There is anyway to solve this?

**Comment by tsw-attila** [comment]
I believe the solution will be to generate and authorize a **refresh_token**. I have no idea how to get this done. 
Looking forward to receiving some help here.

**Comment by tsw-attila** [comment]
Hi @j-oflaherty and @alefgoto have you found a solution?
I wish there was a step-by-step guide or video demonstrating this. Would be appreciated.

**Comment by j-oflaherty** [comment]
Hi @tsw-attila. I haven't found a solution to the bearer token expiring. I was able to find a workaround so that the tasks replication was faster but it's not a permanent solution and still faces the same issue if you are importing a lot of data. In other words, it only works if you work incrementally and you don't generate a lot of data between executions.

I've pushed the changes [here](https://github.com/j-oflaherty/tap-asana). 

As a reference, i've just run it for the first time since the 25 of march and took 15 to complete. Hope this works for you, though the fix for the bearer token is still needed.
